### DISCUSSION TITLE
Using varsFrom

### DISCUSSION THREAD

**Discussion by artem-nefedov** [flux-iac/tofu-controller#299] [Q&A] [answered]
Is it possible to use `varsFrom` to pass non-string variables (object, list, bool, etc.)? And if yes, how you should format the values in a Secret/ConfigMap?

**Comment by chanwit** [comment, marked as answer]
It accepts value in JSON format (that's why a string is also valid). You can use JSON to encode any values you want.

https://github.com/weaveworks/tf-controller/blob/fd0dcd3cc716962ea281b617aac7b0fe083ee607/runner/server.go#L332-L337

and they all are written down into `generated.auto.tfvars.json` file:

https://github.com/weaveworks/tf-controller/blob/fd0dcd3cc716962ea281b617aac7b0fe083ee607/runner/server.go#L404-L405

**Reply by artem-nefedov** [reply]
Sorry, some things are still not clear to me, hope you can elaborate.

1. If ConfigMap has `foo: "123"` in data, will it be passed as a string, or as a number? And is there any way to force the type?
2. If ConfigMap has `bar: '{"aaa": "bbb"}'` in data, will that be passed as a literal string, or as an object? Once again, can I force the type?

**Reply by chanwit** [reply]
1. A number. You would try `foo: "\"123\"".
2. Yes, it becomes an object.

JSON supports three types, string, number, and boolean. To force the type of a value to a string, you would try escaping double quotes for it.

**Reply by BuckinghamIO** [reply]
I've tested this multiple times and passing something like 

```bar: '{"aaa": "bbb"}' ``` 

Still results in tf-controller treaing it as a string and then resulting in it complaining that the value isn't a object that terraform expects.
